java/src/main/java/dojo/supermarket/model/Teller.java

```diff
@@ -0,0 +1,34 @@
+package dojo.supermarket.model;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Teller {
+
+    private final SupermarketCatalog catalog;
+    private final Map<Product, Offer> offers = new HashMap<>();
+
+    public Teller(SupermarketCatalog catalog) {
+        this.catalog = catalog;
+    }
+
+    public void addSpecialOffer(SpecialOfferType offerType, Product product, double argument) {
+        offers.put(product, new Offer(offerType, product, argument));
+    }
+
+    public Receipt checksOutArticlesFrom(ShoppingCart theCart) {
```

Comment:
naming of the variables should be simple, cart instead of theCart will be enough descriptive
```suggestion
    public Receipt checksOutArticlesFrom(ShoppingCart cart) {
```